### PR TITLE
update nft data

### PR DESCRIPTION
**Pull request by miiu96**
Update `attributes` and `uris` for a NFT or SFT

### CODE REVIEW

File: converters/tokenMetaData.go [hunk 1]
```diff
@@ -59,3 +62,32 @@ func whiteListedStorage(uris [][]byte) bool {
...
+		metaData := []byte(fmt.Sprintf(`{"update":{"_id":"%s", "_type": "_doc"}}%s`, id, "\n"))
+		base64Attr := base64.StdEncoding.EncodeToString(nftUpdate.NewAttributes)
+		serializedData := []byte(fmt.Sprintf(`{"script": {"source": "if (ctx._source.containsKey('data')) {ctx._source.data.attributes = params.attributes}","lang": "painless","params": {"attributes": "%s"}}, "upsert": {}}`, base64Attr))
+		if len(nftUpdate.URIsToAdd) != 0 {
+			marshalizedURIS, err := json.Marshal(nftUpdate.URIsToAdd)
+			if err != nil {
+				return err
+			}
+
+			serializedData = []byte(fmt.Sprintf(`{"script": {"source": "if (ctx._source.containsKey('data')) { if (!ctx._source.data.containsKey('uris')) { ctx._source.data.uris = params.uris; } else {  ctx._source.data.uris.addAll(params.uris); }}","lang": "painless","params": {"uris": %s}},"upsert": {}}`, marshalizedURIS))
```
Comment (bogdan-rosianu) [inline converters/tokenMetaData.go:83]: so there is any URI, then `serializedData` will be overwritten. this means:
if hasAttributes && hasURIs { setURIs }
if hasAttributes && !hasURIs { setAttributes }

is this ok ?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: converters/tokenMetaData.go (hunk 1 above)
Comment (miiu96) [reply converters/tokenMetaData.go:83]: if we have URIs ---> will be added to the current list
if we have attributes ---> will be overwritten

File: process/accounts/serialize.go [hunk 2]
```diff
@@ -29,13 +29,30 @@ func (ap *accountsProcessor) SerializeNFTCreateInfo(tokensInfo []*data.TokenInfo
...
+		err = buffSlice.PutData(meta, serializedData)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return buffSlice.Buffers(), nil
+}
+
+func (ap *accountsProcessor) SerializeAccountsESDT(
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/accounts/serialize.go:49]: added.

File: process/logsevents/logsData.go [hunk 3]
```diff
@@ -7,15 +7,16 @@ import (
...
+	timestamp       uint64
+	tokens          data.TokensHandler
+	tagsCount       data.CountTags
+	accounts        data.AlteredAccountsHandler
+	txsMap          map[string]*data.Transaction
+	scrsMap         map[string]*data.ScResult
+	scDeploys       map[string]*data.ScDeployInfo
+	delegators      map[string]*data.Delegator
+	tokensInfo      []*data.TokenInfo
+	updatesNFTsData []*data.UpdateNFTData
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/logsevents/logsData.go:19]: renamed.

**Review by dragos-rebegea — DISMISSED** [review]

File: process/accounts/serialize.go [hunk 4]
```diff
@@ -46,9 +46,10 @@ func (ap *accountsProcessor) SerializeAccounts(accounts map[string]*data.Account
 	return buffSlice.Buffers(), nil
 }
 
+// SerializeAccountsESDT will serialize the provided accounts and nfts updates  in a way that Elasticsearch expects a bulk request
```
Comment (bogdan-rosianu) [inline process/accounts/serialize.go:49]: there are 2 blank spaces between `updates` and `in`

**Review by bogdan-rosianu — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**